client-runtime/src/main/java/com/microsoft/rest/RestClient.java

```diff
@@ -6,241 +6,191 @@
-import okhttp3.ConnectionPool;
-import okhttp3.Interceptor;
-import okhttp3.JavaNetCookieJar;
-import okhttp3.OkHttpClient;
-import okio.AsyncTimeout;
-import retrofit2.Retrofit;
-import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
-
-import java.net.CookieManager;
-import java.net.CookiePolicy;
-import java.net.Proxy;
-import java.util.concurrent.Executor;
+import com.microsoft.rest.v2.http.HttpClient;
+import com.microsoft.rest.v2.http.RxNettyAdapter;
+import com.microsoft.rest.v2.policy.CredentialsPolicy;
+import com.microsoft.rest.v2.policy.LoggingPolicy;
+import com.microsoft.rest.v2.policy.RequestPolicy;
+import com.microsoft.rest.v2.policy.RequestPolicyChain;
+import com.microsoft.rest.v2.policy.RetryPolicy;
+import com.microsoft.rest.v2.policy.SendRequestPolicyFactory;
+import com.microsoft.rest.v2.policy.UserAgentPolicy;
+
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 
 /**
- * An instance of this class stores the client information for making REST calls.
+ * An instance of this class stores configuration for setting up specific service clients.
  */
 public final class RestClient {
-    /** The {@link okhttp3.OkHttpClient} object. */
-    private final OkHttpClient httpClient;
-    /** The {@link retrofit2.Retrofit} object. */
-    private final Retrofit retrofit;
-    /** The original builder for this rest client. */
-    private final RestClient.Builder builder;
-
-    private RestClient(OkHttpClient httpClient,
-                       Retrofit retrofit,
-                       RestClient.Builder builder) {
-        this.httpClient = httpClient;
-        this.retrofit = retrofit;
-        this.builder = builder;
+
+    private final HttpClient httpClient;
+    private final String baseURL;
+    private final String userAgent;
+    private final long readTimeoutMillis;
+    private final long connectionTimeoutMillis;
+    private final SerializerAdapter<?> serializerAdapter;
+    private final ResponseBuilder.Factory responseBuilderFactory;
+    private final ServiceClientCredentials credentials;
+    private LogLevel logLevel;
+
+    private final List<RequestPolicy.Factory> customPolicyFactories;
+    private final RequestPolicyChain fullPolicyChain;
+
+    private RestClient(RestClient.Builder builder) {
+        this.httpClient = builder.httpClient;
+        this.baseURL = builder.baseUrl;
+        this.userAgent = builder.userAgent;
+        this.readTimeoutMillis = builder.readTimeoutMillis;
+        this.connectionTimeoutMillis = builder.connectionTimeoutMillis;
+        this.serializerAdapter = builder.serializerAdapter;
+        this.responseBuilderFactory = builder.responseBuilderFactory;
+        this.credentials = builder.credentials;
+        this.logLevel = builder.logLevel;
+
+        this.customPolicyFactories = builder.customPolicyFactories;
+        this.fullPolicyChain = createPolicyChain();
+    }
+
+    private RequestPolicyChain createPolicyChain() {
```

Comment:
Why is this called RequestPolicyChain? I understand the chain idea, but what are the "policies"?